lib/src/mock_query.dart

```diff
@@ -71,59 +71,71 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
-          final shouldSort = index == 0 ||
-              d1.get(fields[index - 1]) == d2.get(fields[index - 1]);
-          if (!shouldSort) {
-            return 0;
-          }
+      final directions = (parameters['orderedByDirection'] ?? []);
 
-          dynamic value1;
-          if (field is String) {
-            try {
-              value1 = d1.get(field) as Comparable;
-            } catch (error) {
-              // This catch catches the case when the key/value does not exist
-              // and the case when the value is null, and as a result not a
-              // Comparable.
-              value1 = null;
-            }
-          } else if (field == FieldPath.documentId) {
-            value1 = d1.id;
-          }
-          dynamic value2;
-          if (field is String) {
-            try {
-              value2 = d2.get(field);
-            } catch (error) {
-              // This catch catches only the case when the key/value does not
-              // exist.
-              value2 = null;
-            }
-          } else if (field == FieldPath.documentId) {
-            value2 = d2.id;
-          }
-          if (value1 == null && value2 == null) {
-            return 0;
-          }
-          // Return null values first.
-          if (value1 == null) {
-            return -1;
+      if (fields.isEmpty) {
+        return sortedList;
+      }
+
+      int doCompare(dynamic field, bool descending, DocumentSnapshot<T> d1,
+          DocumentSnapshot<T> d2) {
+        dynamic value1;
+        if (field is String) {
+          try {
+            value1 = d1.get(field) as Comparable;
+          } catch (error) {
+            // This catch catches the case when the key/value does not exist
+            // and the case when the value is null, and as a result not a
+            // Comparable.
+            value1 = null;
           }
-          if (value2 == null) {
-            return 1;
+        } else if (field == FieldPath.documentId) {
+          value1 = d1.id;
+        }
+        dynamic value2;
+        if (field is String) {
+          try {
+            value2 = d2.get(field);
+          } catch (error) {
+            // This catch catches only the case when the key/value does not
+            // exist.
+            value2 = null;
           }
-          final compare = value1.compareTo(value2);
-          return descending ? -compare : compare;
-        });
+        } else if (field == FieldPath.documentId) {
+          value2 = d2.id;
+        }
+        if (value1 == null && value2 == null) {
+          return 0;
+        }
+        // Return null values first.
+        if (value1 == null) {
+          return -1;
+        }
+        if (value2 == null) {
+          return 1;
+        }
+        final compare = value1.compareTo(value2);
+        return descending ? -compare : compare;
       }
+
+      sortedList.sort((d1, d2) {
+        var compare = doCompare(fields.first, directions.first, d1, d2);
```

Comment:
I guess it's just a matter of preference. This could work too and is sligthly more readable:
```dart
for (var i = 0; i < fields.length; i++) {
  final compare = doCompare(fields[i], directions[i], d1, d2);
  if (compare != 0) {
    return compare;
  }
}
return 0;
```

Reply:
I ended up making the change at https://github.com/atn832/fake_cloud_firestore/commit/57311c7816163cab22ee2e2686083ae6e1c385ad.